app/Http/Controllers/DiaryController.php

```diff
@@ -23,15 +21,41 @@ public function index()
      */
     public function create()
     {
-        //
+        return view('diary-post');
     }
 
     /**
      * Store a newly created resource in storage.
      */
     public function store(Request $request)
     {
-        //
+        // バリデーション
+        $request->validate([
+            'title' => 'required|string|max:50',
+            'description' => 'required|string|max:255',
+            'rating' => 'required|integer|min:1|max:5',
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif|max:2048',
+        ]);
+
+        // 画像の保存
+        $imagePath = null;
+        if ($request->hasFile('image')) {
+            $image = $request->file('image'); // アップロードされた画像を取得
+            $imagePath = $image->store('images', 'public'); // 画像を保存し、そのパスを取得
+        }
+        // データの保存
+        $diary = new Diary();
+
+        $diary->user_id = auth()->id(); // ログインしているユーザーのIDを設定
+        $diary->image_path = $imagePath;
+        $diary->title = $request->title;
+        $diary->comment = $request->description;
+        $diary->rating = $request->rating;
+
+        $diary->save();
```

Comment:
こちら、save()をsaveOrFail()にしてみましょう！
保存に失敗した時にきちんとエラーハンドリングすることができます！例えばtry catchと組み合わせて以下のように使用することが多いです。
```suggestion
        try {
            $diary = new Diary();
    
            $diary->user_id = auth()->id(); // ログインしているユーザーのIDを設定
            $diary->image_path = $imagePath;
            $diary->title = $request->title;
            $diary->comment = $request->description;
            $diary->rating = $request->rating;
    
            $diary->save();
        } catch (Exception $e) {
        //エラーメッセージを記述
        }
```

Reply:
@hipomaru 

a0af2a784163f75566ed896c175dfeddfac450e9

上記のコミットで対応しました

Reply:
@hipomaru 

b1c0e8d625c36c18bf109affbe2ed260c40ecb35

上記のコミットで対応しました